src/utils/integClient.ts

```diff
@@ -0,0 +1,439 @@
+import * as errors from './error';
+import { fieldValues, initFormsPromise, initInfo } from './init';
+import { encodeGetParams } from './primitives';
+import { parseError } from './error';
+import { API_URL } from './client';
+
+const TYPE_MESSAGES_TO_IGNORE = [
+  // e.g. https://sentry.io/organizations/feathery-forms/issues/3571287943/
+  'Failed to fetch',
+  // e.g. https://sentry.io/organizations/feathery-forms/issues/3529742129/
+  'Load failed'
+];
+
+export default class IntegClient {
+  formKey: string;
+  version?: string;
+  noSave?: boolean;
+  ignoreNetworkErrors: any; // this should be a ref
+  draft: boolean;
+  bypassCDN: boolean;
+  submitQueue: Promise<any>;
+  constructor(
+    formKey = '',
+    ignoreNetworkErrors?: any,
+    draft = false,
+    bypassCDN = false
+  ) {
+    this.formKey = formKey;
+    this.ignoreNetworkErrors = ignoreNetworkErrors;
+    this.draft = draft;
+    this.bypassCDN = bypassCDN;
+    this.submitQueue = Promise.resolve();
+  }
+
+  async _checkResponseSuccess(response: any) {
+    let payload;
+    switch (response.status) {
+      case 200:
+      case 201:
+        return;
+      case 400:
+        payload = JSON.stringify(await response.clone().text());
+        console.error(payload.toString());
+        return;
+      case 401:
+        throw new errors.SDKKeyError();
+      case 404:
+        throw new errors.FetchError("Can't find object");
+      case 409:
+        location.reload();
+        return;
+      case 500:
+        throw new errors.FetchError('Internal server error');
+      default:
+        throw new errors.FetchError('Unknown error');
+    }
+  }
+
+  _fetch(url: any, options: any, parseResponse = true) {
+    const { sdkKey } = initInfo();
+    const { headers, ...otherOptions } = options;
+    options = {
+      cache: 'no-store',
+      // Write requests must succeed so data is tracked
+      keepalive: ['POST', 'PATCH', 'PUT'].includes(options.method),
+      headers: {
+        Authorization: 'Token ' + sdkKey,
+        ...headers
+      },
+      ...otherOptions
+    };
+    return fetch(url, options)
+      .then(async (response) => {
+        if (parseResponse) await this._checkResponseSuccess(response);
+        return response;
+      })
+      .catch((e) => {
+        // Ignore TypeErrors if form has redirected because `fetch` in
+        // Safari will error after redirect
+        if (
+          (this.ignoreNetworkErrors?.current ||
+            TYPE_MESSAGES_TO_IGNORE.includes(e.message)) &&
+          e instanceof TypeError
+        )
+          return;
+        throw e;
+      });
+  }
+
+  // THIRD-PARTY INTEGRATIONS
```

Comment:
This comment can maybe be moved to the top of the class definition to describe the purpose of this client